specs/BlockchainLayer/EpochManager/Epoch.md

```diff
@@ -0,0 +1,41 @@
+# Epoch
+
+All blocks are split into epochs. Within one epoch, the set of validators is fixed, and validator rotation
+happens at epoch boundaries.
+
+Genesis block is in its own epoch. After that, a block is either in its parent's epoch or
+starts a new epoch if it meets certain conditions.
+
+### BlockHeight
+Each block has a height. Genesis block has height 0, the height of each block is greater than
```

Comment:
The genesis block is not necessarily zero?

Reply:
Good point. Somehow I didn't notice this :(